pyamg/amg_core/ruge_stuben.h

```diff
@@ -956,345 +943,1339 @@ void cr_helper(const I A_rowptr[],  const int A_rowptr_size,
     }
 }
 
+/* First pass of classical AMG interpolation to build row pointer for P based
+ * on SOC matrix and CF-splitting. Same method used for standard and modified
+ * AMG interpolation below. 
+ *
+ * Parameters:
+ * -----------
+ *      n_nodes : const int
+ *          Number of rows in A
+ *      Sp : const array<int>
+ *          Row pointer for SOC matrix, C
+ *      Sj : const array<int>
+ *          Column indices for SOC matrix, C
+ *      splitting : const array<int>
+ *          Boolean array with 1 denoting C-points and 0 F-points
+ *      Pp : array<int>
+ *          empty array to store row pointer for matrix P
+ *
+ * Returns:
+ * --------
+ * Nothing, Pp is modified in place. 
+ *
+ */
+template<class I>
+void rs_standard_interpolation_pass1(const I n_nodes,
+                                     const I Sp[], const int Sp_size,
+                                     const I Sj[], const int Sj_size,
+                                     const I splitting[], const int splitting_size,
+                                           I Pp[], const int Pp_size)
+{
+    I nnz = 0;
+    Pp[0] = 0;
+    for (I i = 0; i < n_nodes; i++){
+        if( splitting[i] == C_NODE ){
+            nnz++;
+        }
+        else {
+            for (I jj = Sp[i]; jj < Sp[i+1]; jj++){
+                if ( (splitting[Sj[jj]] == C_NODE) && (Sj[jj] != i) )
+                    nnz++;
+            }
+        }
+        Pp[i+1] = nnz;
+    }
+}
+
+
+/* Produce the classical "standard" AMG interpolation operator. The first pass
+ * uses the strength of connection matrix and C/F splitting to compute the row
+ * pointer for the prolongator. The second pass fills in the nonzero entries of
+ * the prolongator. Formula can be found in Eq. (3.8) in [1].
```

Comment:
@bensworth  I don't see an equation (3.8) in 
https://onlinelibrary.wiley.com/doi/abs/10.1002/nla.559 
Is this the right paper for reference [1]?